velox/vector/tests/VectorMakerTest.cpp

```diff
@@ -458,6 +459,55 @@ TEST_F(VectorMakerTest, arrayVectorFromJson) {
   }
 }
 
+TEST_F(VectorMakerTest, nestedArrayVectorFromJson) {
+  auto arrayVector = maker_.nestedArrayVectorFromJson<int32_t>({
+      "[[1, 2], [2, 3, 4], [null, 7]]",
+      "[[1, 3, 7, 9], []]",
+      "[]",
+      "[null]",
+      "[[1, 2, null], [], null]",
+  });
+
+  VectorPtr expectedVector = maker_.arrayVectorNullable<int32_t>({
+      {{1, 2}},
+      {{2, 3, 4}},
+      {{std::nullopt, 7}},
+      {{1, 3, 7, 9}},
+      {{}},
+      {{}},
+      {{std::nullopt}},
+      {{1, 2, std::nullopt}},
+      {{}},
+      std::nullopt,
+  });
+
+  VectorPtr elementsOfArrayVector = arrayVector->elements();
+
+  EXPECT_EQ(*arrayVector->elements()->type(), *ARRAY(INTEGER()));
```

Comment:
Why is * needed here ? EXPECT(arrayVector->elements()->type() == ARRAY(INTEGER())); should work I think.

Reply:
without the * indirection, this check would fail as the pointers have different addresses but they seem to point to the same stuff so the check passes with *